ea-2022-04-ndvi-automation.ipynb

```diff
@@ -0,0 +1,1600 @@
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "            mean_ndvi  site\n",
+       "2017-03-17   0.281132  HARV"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create dataframe of mean NDVI in this cell using the functions created above\n",
+    "# Important: to use the ungraded tests below as a sanity check,\n",
+    "# name your columns: mean_ndvi and site\n",
+    "# Call the dataframe at the end of the cell so the tests run on it!\n",
+    "# Be sure that the date column is an index of type date\n",
+    "# HINT: the time series lessons may help you remember how to do this!\n",
+    "\n",
+    "# Define paths to data and crop geodataframe files\n",
+    "landsat_dirpath_harv = os.path.join(\"ndvi-automation\",\n",
+    "                                    \"sites\",\n",
+    "                                    \"HARV\",\n",
+    "                                    \"landsat-crop\",\n",
+    "                                    \"LC080130302017031701T1-SC20181023151837\",\n",
+    "                                    \"*band[4-5]*.tif\")\n",
+    "crop_dirpath_harv = os.path.join(\"ndvi-automation\",\n",
+    "                                 \"sites\",\n",
+    "                                 \"HARV\",\n",
+    "                                 \"vector\",\n",
+    "                                 \"HARV-crop.shp\")\n",
+    "\n",
+    "crop_dirpath_sjer = os.path.join(\"ndvi-automation\",\n",
+    "                                 \"sites\",\n",
+    "                                 \"SJER\",\n",
+    "                                 \"vector\",\n",
+    "                                 \"SJER-crop.shp\")\n",
+    "\n",
+    "# Open crop files for both sites with geopandas\n",
+    "harv_crop = gpd.read_file(crop_dirpath_harv)\n",
+    "sjer_crop = gpd.read_file(crop_dirpath_sjer)\n",
+    "\n",
+    "# Sort list of .tif files for single date in HARV landsat data\n",
+    "landsat_paths_harv = sorted(glob(landsat_dirpath_harv))\n",
+    "\n",
+    "# Open, clean, and crop data for single date with process_bands\n",
+    "landsat_harv_march = process_bands(\n",
+    "    landsat_paths_harv, crop_dirpath_harv, stack=True)\n",
+    "\n",
+    "# Calculate average NDVI for this date with normalized_diff\n",
+    "landsat_harv_ndvi = normalized_diff(\n",
+    "    landsat_harv_march[1], landsat_harv_march[0])\n",
+    "\n",
+    "# Save NDVI value, date, and site name to pandas df\n",
+    "ndvi_values = [landsat_harv_ndvi]\n",
+    "\n",
+    "date_single = '20170317'\n",
+    "datetime_object_single = [datetime.strptime(date_single, '%Y%m%d')]\n",
+    "\n",
+    "df_harv_single_date = pd.DataFrame(\n",
+    "    {'mean_ndvi': ndvi_values}, index=datetime_object_single)\n",
+    "df_harv_single_date['site'] = 'HARV'\n",
+    "df_harv_single_date"
+   ]
+  },
+  {
```

Comment:
Love all of your commented steps. It makes it very easy to read and work through. Aids in reproducibility!

Reply:
Thanks!